builder/amazon/common/ami_config_test.go

```diff
@@ -29,26 +34,52 @@ func TestAMIConfigPrepare_name(t *testing.T) {
 	}
 }
 
+type mockEC2Client struct {
+	ec2iface.EC2API
+}
+
+func (m *mockEC2Client) DescribeRegions(*ec2.DescribeRegionsInput) (*ec2.DescribeRegionsOutput, error) {
```

Comment:
This is a useful pattern, likely it could be used in lots of other places too in the AWS providers. If the team is willing to adopt `testify` as a dependency it has some helpful tools for building mocks like this in a more fluent manner.

Reply:
Agreed. I've been meaning to tinker with these interface abstractions for a while, but this is the first concrete chance I've had. Moving to using these mocks should also help us make better modular and testable code.